src/praisonai/praisonai/gateway/server.py

```
@@ -2427,59 +2459,419 @@ async def stop_channels(self) -> None:
+            channels_cfg: Full channels configuration
+        """
+        logger.info(f"Restarting channel '{channel_name}'...")
+        
+        # Find and cancel the existing channel task
+        if channel_name in self._channel_bots:
+            # Stop the bot
+            bot = self._channel_bots[channel_name]
+            try:
+                if hasattr(bot, "stop"):
+                    await bot.stop()
+            except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)